### PR TITLE
Fix for #5.

### PR DESCRIPTION
**Pull request by jbschiratti**
Fixing #5.

### CODE REVIEW

File: mne_features/feature_extraction.py [hunk 1]
```diff
@@ -31,10 +31,15 @@ class FeatureFunctionTransformer(FunctionTransformer):
         If True, the array X will be checked before calling the function.
         If possible, a 2d Numpy array is returned. Otherwise, an exception
         will be raised. If False, the array X is not checked.
+
+    kw_args : dict or None (default: None)
+        If not None, dictionary of additional keyword arguments to pass to the
+        feature function.
     """
-    def __init__(self, func=None, validate=True):
+    def __init__(self, func=None, validate=True, kw_args=None):
```
Comment (agramfort) [inline mne_features/feature_extraction.py]: call kw_args params as in sklearn

File: mne_features/feature_extraction.py [hunk 2]
```diff
@@ -64,6 +69,40 @@ def get_feature_names(self):
...
+            # If `_params['func'] is an actual Python function
+            _to_inspect = _params['func']
+        args, _, _ = getargs(_to_inspect.func_code)
+        defaults = _to_inspect.func_defaults
+        if defaults is None:
+            return dict()
+        else:
+            n_defaults = len(defaults)
+            func_params = {key: value for key, value in
+                           zip(args[-n_defaults:], defaults)}
```
Comment (agramfort) [inline mne_features/feature_extraction.py]: get_params should use self.kw_args and overwrite the inspected parameters if present in self.kw_args

**Comment by agramfort** [comment]
can you start making some examples to use the added functions? they should start with plot_ in the examples folder. thanks

File: mne_features/feature_extraction.py (hunk 1 above)
Comment (jbschiratti) [reply mne_features/feature_extraction.py]: kw_args seems to be used the same way in https://github.com/scikit-learn/scikit-learn/blob/a24c8b464d094d2c468a16ea9f8bf8d42d949f84/sklearn/preprocessing/_function_transformer.py#L71. What am I missing ?

File: mne_features/feature_extraction.py (hunk 2 above)
Comment (jbschiratti) [reply mne_features/feature_extraction.py]: Instead of get_params, set_params should do that, shouldn't it ?

Comment (agramfort) [reply mne_features/feature_extraction.py]: what should set_params do?

Comment (jbschiratti) [reply mne_features/feature_extraction.py]: Given a dict (`param: value`) it should set the params of the function in the transformer (kw_args).

**Comment by agramfort** [comment]
the transformer should on the fly take the params in self.params and
eventually overwrite the default parameters

**Comment by agramfort** [comment]
are you testing anywhere that set_params and get_params work as expected? I would test the object directly not just via extract features.

**Comment by jbschiratti** [comment]
+1. Indeed, I should have tested `FeatureFunctionTransformer` (and its methods) separately.

File: mne_features/feature_extraction.py [hunk 3]
```diff
@@ -64,6 +70,53 @@ def get_feature_names(self):
...
+        if self.params is not None:
+            func_params.update(self.params)
+        return func_params
+
+    def set_params(self, **new_params):
+        """ Set the parameters of the given feature function. """
+        valid_params = self.get_params()
+        for key in new_params.keys():
+            if key not in valid_params.keys():
```
Comment (agramfort) [inline mne_features/feature_extraction.py]: `if key not in valid_params:`

is much better and faster.

**Comment by agramfort** [comment]
thx @jbschiratti

**Pull request merged**